src/js/routable.js

```diff
@@ -11,6 +11,46 @@ var gpii  = fluid.registerNamespace("gpii");
 
 fluid.registerNamespace("gpii.express.routable");
 
+// TODO: Discuss if / how / when to pull this up into Infusion itself.
+/**
+ *
+ * Return a map of immediate child components that extend `grade`.
+ *
+ * @param {Object} that - The component itself.
+ * @param {String} grade - The gradename to search for.
+ * @return {Object} - A map of child components that extend `grade`, keyed by member name.
+ *
+ */
+gpii.express.routable.childrenWithGrade = function (that, grade) {
+    var componentsMatchingGrade = {};
+    fluid.visitComponentChildren(that, function (childComponent, name) {
+        if (fluid.componentHasGrade(childComponent, grade)) {
+            componentsMatchingGrade[name] = childComponent;
+        }
+    }, { flat: true });
+    return componentsMatchingGrade;
+};
+
+/**
+ *
+ * Order an array of components using their `namespace` and `priority` options.
+ *
+ * @param unsortedComponentMap {Object} - A map of components to be sorted.
+ * @return {Array} - The components, sorted by namespaced priority.
+ *
+ */
+gpii.express.routable.prioritiseComponentArray = function (unsortedComponentMap)  {
+    var componentOptionsById = {};
+    fluid.each(unsortedComponentMap, function (singleComponent, memberName) {
+        var optionsForPrioritySorting = fluid.copy(singleComponent.options);
+        optionsForPrioritySorting.pocketedComponent = singleComponent;
+        componentOptionsById[memberName] = optionsForPrioritySorting;
+    });
+    var sortedComponentOptions = fluid.parsePriorityRecords(componentOptionsById, "router entry");
+    var sortedComponents = fluid.transform(sortedComponentOptions, function (componentDef) { return componentDef.pocketedComponent; });
```

Comment:
This can just be fluid.getMembers(sortedComponentOptions, "pocketedComponent")

Reply:
Done.